unittests/PhasarLLVM/TypeHierarchy/DIBasedTypeHierarchyTest.cpp

```diff
@@ -0,0 +1,241 @@
+
+#include "phasar/PhasarLLVM/TypeHierarchy/DIBasedTypeHierarchy.h"
+
+#include "phasar/Config/Configuration.h"
+#include "phasar/PhasarLLVM/DB/LLVMProjectIRDB.h"
+#include "phasar/PhasarLLVM/Utils/LLVMShorthands.h"
+#include "phasar/Utils/Utilities.h"
+
+#include "TestConfig.h"
+#include "gtest/gtest.h"
+
+namespace psr {
+
+// Check basic type hierarchy construction
+TEST(DBTHTest, BasicTHReconstruction_1) {
+  LLVMProjectIRDB IRDB(unittest::PathToLLTestFiles +
+                       "type_hierarchies/type_hierarchy_1_cpp_dbg.ll");
+  DIBasedTypeHierarchy DBTH(IRDB);
+
+  const auto &Types = DBTH.getAllTypes();
+  const auto &SubTypes = DBTH.getSubTypes(DBTH.getType("Base"));
+
+  const auto &BaseType = DBTH.getType("Base");
+  ASSERT_NE(nullptr, BaseType);
+
+  EXPECT_TRUE(DBTH.hasType(BaseType));
+  EXPECT_TRUE(DBTH.hasVFTable(BaseType));
+
+  const auto &ChildType = DBTH.getType("Child");
+  ASSERT_NE(nullptr, ChildType);
+
+  EXPECT_TRUE(DBTH.hasType(ChildType));
+  EXPECT_TRUE(SubTypes.find(ChildType) != SubTypes.end());
+  ASSERT_TRUE(DBTH.hasVFTable(ChildType));
+  const auto &VTableForChild = DBTH.getVFTable(ChildType);
+
+  ASSERT_NE(nullptr, VTableForChild);
+  EXPECT_TRUE(VTableForChild->getFunction(0)->getName() == "_ZN5Child3fooEv");
+}
+
+TEST(DBTHTest, BasicTHReconstruction_2) {
+  LLVMProjectIRDB IRDB({unittest::PathToLLTestFiles +
+                        "type_hierarchies/type_hierarchy_17_cpp_dbg.ll"});
```

Comment:
the indices are inconsistent, that's very surprising and confusing. Is there a reason you omit the other cases? What about case 16 (ll index) ?